### ISSUE TITLE
AsyncServer: client able to connect to non-declared namespaces

### ISSUE THREAD

**Issue by simingy** [miguelgrinberg/python-socketio#463]
in reference implementation, if a namespace is not declared on the server side,
it gets rejected:

(node.js socket.io server, AsyncClient in python)
```
DEBUG:asyncio:Using selector: KqueueSelector
INFO:engineio.client:Attempting WebSocket connection to ws://localhost:3000/socket.io/?transport=websocket&EIO=3
INFO:engineio.client:WebSocket connection accepted with {'sid': 'xL5TuqhMYmR-mcxUAAAA', 'upgrades': [], 'pingInterval': 25000, 'pingTimeout': 5000}
INFO:socketio.client:Engine.IO connection established
INFO:engineio.client:Attempting WebSocket connection to ws://localhost:3000/socket.io/?transport=websocket&EIO=3
INFO:engineio.client:Sending packet PING data None
INFO:engineio.client:Received packet MESSAGE data 0
INFO:socketio.client:Namespace / is connected
connection established 1586874034.752005
INFO:engineio.client:Received packet PONG data None
INFO:engineio.client:WebSocket connection accepted with {'sid': 'xYttzOjM2NF5noAKAAAB', 'upgrades': [], 'pingInterval': 25000, 'pingTimeout': 5000}
INFO:socketio.client:Engine.IO connection established
INFO:engineio.client:Sending packet PING data None
INFO:engineio.client:Received packet MESSAGE data 0
INFO:socketio.client:Namespace / is connected
INFO:engineio.client:Sending packet MESSAGE data 0/abc
INFO:engineio.client:Received packet PONG data None
INFO:engineio.client:Received packet MESSAGE data 4/abc,"Invalid namespace"
INFO:socketio.client:Connection to namespace /abc was rejected
```

where as in `AsyncServer` it does get connected: (same client in Python)
```
DEBUG:asyncio:Using selector: KqueueSelector
INFO:engineio.client:Attempting WebSocket connection to ws://localhost:8080/socket.io/?transport=websocket&EIO=3
INFO:engineio.client:WebSocket connection accepted with {'sid': '65228b1f6df74d68890e3c77aa4a72f6', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
INFO:socketio.client:Engine.IO connection established
INFO:engineio.client:Attempting WebSocket connection to ws://localhost:8080/socket.io/?transport=websocket&EIO=3
INFO:engineio.client:Sending packet PING data None
INFO:engineio.client:Received packet MESSAGE data 0
INFO:socketio.client:Namespace / is connected
connection established 1586874166.1954072
INFO:engineio.client:Received packet PONG data None
INFO:engineio.client:WebSocket connection accepted with {'sid': '0c733cb941ab4f4ead5e762207b53ab8', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
INFO:socketio.client:Engine.IO connection established
INFO:engineio.client:Sending packet PING data None
INFO:engineio.client:Received packet MESSAGE data 0
INFO:socketio.client:Namespace / is connected
INFO:engineio.client:Sending packet MESSAGE data 0/abc
INFO:engineio.client:Received packet PONG data None
INFO:engineio.client:Received packet MESSAGE data 0/abc
INFO:socketio.client:Namespace /abc is connected
```

**Comment by miguelgrinberg** [comment]
What version of the server are you using? This was fixed in release 4.4.0.

**Comment by simingy** [comment]
@miguelgrinberg 

python-engineio   3.12.1
python-socketio   4.5.1

Python 3.6.10

**Comment by miguelgrinberg** [comment]
Oh, I see, so you are attempting to connect to the namespace, not send an event to it. What was implemented in 4.4.0 is a check in the emit() method that prevents you from emitting to a namespace that isn't connected.

Okay, I can look at this one as well, though it is a lot less critical in my opinion.

**Comment by miguelgrinberg** [comment]
Duplicate of https://github.com/miguelgrinberg/python-socketio/issues/822. Closing.
